File: iOS/Peliplus/model/Movie.swift
//
//  Movie.swift
//  Peliplus
//
//  Created by Francisco Sanchez on 29/04/2020.
//  Copyright © 2020 Francisco Sanchez. All rights reserved.
//

import Foundation
import UIKit

public class Movie:NSObject {
    var movieId:Int!
    var name:String!  //implicity unwrapped optional, he is not going to check if null
    var image:String!
    var years:Int!
    var descrip:String!
    //var image:UIImage!
    var typeMovie:String!
    var typeMovieId:Int!
    
    init(name:String) {
        self.name=name
       	
    }
    
    init(cMovie:CMovie) {
        self.movieId = Int(cMovie.movieId)
        self.name=cMovie.name
        self.image=cMovie.image
        self.years=Int(cMovie.year)
        self.descrip=cMovie.descrip
        self.typeMovie=cMovie.typeMovie
        self.typeMovieId=Int(cMovie.typeMovieId)
        //self.image=#imageLiteral(resourceName: "avengers-inifity-war")
    }
    
    static func cMoviesToMovies(cMovies:[CMovie])->[Movie]{
        var movies:[Movie]=[]
        
        for cMovie in cMovies{
            
            movies.append(Movie(cMovie:cMovie))
        }
        
        return movies
    }
    
}

File: Android/PeliPlus/app/src/main/java/com/peliplus/adr/model/Movie.kt
package com.peliplus.adr.model

class Movie {
    var movie_id: Int = 0
    var name: String = ""
    var year: Int = 0
    var description: String = ""
    var image: String = ""
    var type_movie_id: Int = 0
    var type_movie: String = ""
    var favorite : Favorite? = null
    var average : Float = 0.0F //No precisión needed
    var qualification : Qualification? = null
    var favorites = ArrayList<Favorite>()
    var qualifications = ArrayList<Qualification>()
}